node/router/src/lib.rs

```diff
@@ -552,14 +558,14 @@ impl<N: Network> Router<N> {
 
     /// Removes the connected peer and adds them to the candidate peers.
     pub fn remove_connected_peer(&self, peer_ip: SocketAddr) {
-        // Removes the bidirectional map between the listener address and (ambiguous) peer address.
-        self.resolver.remove_peer(&peer_ip);
         // Remove this peer from the connected peers, if it exists.
         self.connected_peers.write().remove(&peer_ip);
```

Comment:
Would it make sense to hold the lock throughout the function, so we update `connected_peers` and `candidate_peers` atomically?

Reply:
Actually, it seems like we can't currently do this due to deadlock potential; however, I'm open to tackling this in a follow-up.

Reply:
Sounds good!